certora/applyHarnessFIFO.patch

```diff
@@ -0,0 +1,250 @@
++        return dll.getValueOf(_id);
++    }
++
++    function getHead() public view returns (address) {
++        return dll.getHead();
++    }
++
++    function getTail() public view returns (address) {
++        return dll.getTail();
++    }
++
++    function getNext(address _id) public view returns (address) {
++        return dll.getNext(_id);
++    }
++
++    function getPrev(address _id) public view returns (address) {
++        return dll.getPrev(_id);
++    }
++
++    function remove(address _id) public {
++        dll.remove(_id);
++    }
++
++    function insertSorted(
++        address _id,
++        uint256 _value,
++        uint256 _maxIterations
++    ) public {
++        dll.insertSorted(_id, _value, _maxIterations);
++    }
++
++    // SPECIFICATION HELPERS
++
++    function getInsertedAfter() public view returns (address) {
++        return dll.insertedAfter;
++    }
++
++    function getInsertedBefore() public view returns (address) {
++        return dll.insertedBefore;
++    }
++
++    function getLength() public view returns (uint256) {
++        uint256 len;
++        for (address current = getHead(); current != address(0); current = getNext(current)) len++;
++        return len;
++    }
++
++    function linkBetween(address _start, address _end) internal view returns (bool, address) {
++        if (_start == _end) return (true, address(0));
++        for (uint256 maxIter = getLength(); maxIter > 0; maxIter--) {
++            address next = getNext(_start);
++            if (next == _end) return (true, _start);
++            _start = next;
++        }
++        return (false, address(0));
++    }
++
++    function isForwardLinkedBetween(address _start, address _end) public view returns (bool ret) {
++        (ret, ) = linkBetween(_start, _end);
++    }
++
++    function prevFromHead(address _end) public view returns (address last) {
++        (, last) = linkBetween(getHead(), _end);
++    }
++
++    function greaterThanUpTo(
++        uint256 _value,
++        address _to,
++        uint256 _maxIter
++    ) public view returns (bool) {
++        address from = getHead();
++        for (; _maxIter > 0; _maxIter--) {
++            if (from == _to) return true;
++            if (getValueOf(from) < _value) return false;
++            from = getNext(from);
++        }
++        return true;
++    }
++
++    function lenUpTo(address _to) public view returns (uint256) {
++        uint256 maxIter = getLength();
++        address from = getHead();
++        for (; maxIter > 0; maxIter--) {
++            if (from == _to) break;
++            from = getNext(from);
++        }
++        return getLength() - maxIter;
++    }
++}
+diff -ruN test/TestRedBlackBinaryTreeOptimized.sol test/TestRedBlackBinaryTreeOptimized.sol
+--- test/TestRedBlackBinaryTreeOptimized.sol	2022-12-16 20:12:44.271470415 +0100
++++ test/TestRedBlackBinaryTreeOptimized.sol	1970-01-01 01:00:00.000000000 +0100
```

Comment:
No need, I just found it weird to have the test folder in the src folder. I switch back to not removing them in the harness, it will be fixed later if we move the files anyway